specs/bellatrix/fork-choice.md

```diff
@@ -178,7 +178,7 @@ def on_block(store: Store, signed_block: SignedBeaconBlock) -> None:
     assert store.finalized_checkpoint.root == finalized_checkpoint_block
 
     # Check the block is valid and compute the post-state
-    state = pre_state.copy()
+    state = pre_state
```

Comment:
Bellatrix requires the prestate for the merge block check below. You can avoid all other copies. Also why the rename instead of just using pre_state or directly calling state the first copy?

Reply:
I get your point, and it seems for the Bellatrix case, the copy can also be eliminated by simply putting the merge check before `state_transition`. Let me think about a candidate name for `state` to clarify things.

Reply:
After a further thought, I think there's no need to change,`state = pre_state` doesn't actually make a copy, just an alias to hint that `state_transition` is being applied to it.

Reply:
Yeah but the point is why name it pre_state to start with if it's never used?

Reply:
I think maybe because it initially starts from the state of `parent_root`, this naming is just to convey this hint.